etils/epath/abstract_path.py

```diff
@@ -112,6 +112,17 @@ def rglob(self: _T, pattern: str) -> Iterator[_T]:
     """Yields all matching files recursively (of any kind)."""
     return self.glob(f'**/{pattern}')
 
+  @abstractmethod
+  def walk(
+      self: _T,
+      top_down: bool = True,
+      on_error: Optional[Callable | str] = None,
+      follow_symlinks: bool = False,
+      max_depth: Optional[int] = None,
+      **kwargs,
+  ) -> Iterator[_T, Iterable[_T], Iterable[_T]]:
```

Comment:
The fsspec backend accepts a string as argument for `on_error` which is not supported by the official pathlib.pyi signature. The fsspec backend also introduces the `kwargs` and `max_depth` arguments. We can cover these cases as follow
```python
  @abstractmethod
  def walk(
      self,
      top_down: bool = True,
      on_error: Callable[[OSError], object] | str | None = None,
      follow_symlinks: bool = False,
      max_depth: int | None = None,
      **kwargs: Any,
  ) -> Iterator[tuple[Self, list[str], list[str]]]:
```

To prevent the typos actually we can replace  `kwargs` by an optional dict argument called `fsspec_kwargs` and use this dict only in the fsspec backend or we can simply remove  `kwargs` but it will means to loose the ability to control how fsspec list files during the walk. what do you think about it ?

Notes:
- I modify the `maxdepth` args from fsspec to `max_depth` to make it easier to remember for  the users.
- fsspec.walk doc https://filesystem-spec.readthedocs.io/en/latest/api.html?highlight=walk#fsspec.spec.AbstractFileSystem.walk

Reply:
`epath.Path` is supposed to abstract away the backends and filesystem that the path use. The fact we use `fsspec`/`tensorflow` or other is an internal detail that might change in the future.

So I don't think the public API should expose any of `max_depth` and `kwargs`. Otherwise the same code would behave inconsistently across `gcs` / local depending on which backend is used.

Reply:
done